Box.V2/Managers/BoxFilesManager.cs

```diff
@@ -280,24 +280,20 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
         /// <summary>
         /// Upload a part of the file to the session.
         /// </summary>
-        /// <param name="id">The session id.</param>
-        /// <param name="sha1">The message digest of the part body, formatted as specified by RFC 3230.</param>
+        /// <param name="uploadPartUri">Upload Uri from Create Session which include SessionId</param>
+        /// <param name="sha">The message digest of the part body, formatted as specified by RFC 3230.</param>
         /// <param name="partId">A valid 8 character hex string that identifies the part upload request.</param>
         /// <param name="partStartOffsetInBytes">Part begin offset in bytes.</param>
         /// <param name="sizeOfOriginalFileInBytes">Size of original file in bytes.</param>
         /// <param name="stream">The file part stream.</param>
         /// <returns></returns>
-        public async Task<bool> UploadPartAsync(string id, string sha1, string partId, long partStartOffsetInBytes, long sizeOfOriginalFileInBytes, Stream stream)
+        public async Task<bool> UploadPartAsync(Uri uploadPartUri, string sha, string partId, long partStartOffsetInBytes, long sizeOfOriginalFileInBytes, Stream stream)
```

Comment:
Yes, synced up with Karthik today and finalized this signature. Java does it a bit differently since they store the complete state of the response, but agreed on this signature since we have the URL already.